clippy_lints/src/const_static_lifetime.rs

```diff
@@ -0,0 +1,56 @@
+use syntax::ast::{Item, ItemKind, TyKind, ExprKind};
+use rustc::lint::{LintPass, EarlyLintPass, LintArray, EarlyContext};
+use utils::span_help_and_lint;
+
+/// **What it does:** Checks for constants with an explicit `'static` lifetime.
+///
+/// **Why is this bad?** Adding `'static` to every reference can create very complicated types.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+///  const FOO: &'static [(&'static str, &'static str, fn(&Bar) -> bool)] = &[..]
+/// ```
+/// This code can be rewritten as
+/// ```rust
+///  const FOO: &[(&str, &str, fn(&Bar) -> bool)] = &[...]
+/// ```
+
+declare_lint! {
+    pub CONST_STATIC_LIFETIME, 
+    Warn,
```

Comment:
Well.. there was no need to update from try macro invocations to the question mark operator, other than readability. But yes, we should delay making it warn by default for the next few releases. I think 3 releases compatibility is what nursery crates require.